src/Files.App/Utils/Storage/Operations/ShellFilesystemOperations.cs

```diff
@@ -792,8 +806,34 @@ private void CancelOperation(object operationID)
 
 		private async Task<bool> RequestAdminOperation()
 		{
-			// TODO: Implement this method
-			return false;
+			var dialogService = Ioc.Default.GetRequiredService<IDialogService>();
+			return await dialogService.ShowDialogAsync(new ElevateConfirmDialogViewModel()) == DialogResult.Primary;
+		}
+
+		private async Task<IStorageHistory> RunAdminOperationAsync(ShellOperationRequest request)
+		{
+			var req = JsonSerializer.Serialize(request);
+			byte[] buffer = Encoding.UTF8.GetBytes(req);
+			using (MemoryMappedFile mmf = MemoryMappedFile.CreateNew(request.ID, req.Length, MemoryMappedFileAccess.ReadWrite, MemoryMappedFileOptions.None, HandleInheritability.Inheritable))
+			{
+				using (MemoryMappedViewAccessor accessor = mmf.CreateViewAccessor())
+				{
+					accessor.Write(0, req.Length);
+					accessor.WriteArray(sizeof(int), buffer, 0, buffer.Length);
+				}
+
+				using Process process = new Process();
+				process.StartInfo.UseShellExecute = true;
+				process.StartInfo.Verb = "RunAs";
+				process.StartInfo.FileName = Path.Combine(Windows.ApplicationModel.Package.Current.InstalledPath, "Files.App.Elevated", "Files.App.Elevated.exe");
+				process.StartInfo.Arguments = string.Join(" ", "FileOperation", request.ID);
+				await SafetyExtensions.IgnoreExceptions(() =>
+				{
+					process.Start();
+					return process.WaitForExitAsync();
+				});
```

Comment:
I think it might be better to handle the exception there, instead of ignoring it. We would be able to report `Operation failed` in the `Status center`, whenever a user rejects the second popup

Reply:
Good idea. I can also try to return success of the operarion through the return value of the process.